### PR TITLE
Gh-390: Add "What is graph" section to user guide

### LINKED ISSUE #390: What is Graph? Additional section in user guide
Following the restructuring of the documentation under https://github.com/gchq/gaffer-doc/issues/378 a page on Graph should be created to ensure that the end user / developer is aware of the toechnology.

Topics to include:

What is a graph?
How Gaffer uses Graph?

### PR DESCRIPTION
**Pull request by rj77259**

### CODE REVIEW

File: docs/user-guide/gaffer-basics/what-is-a-graph.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+```mermaid
+flowchart TD
+    subgraph Relationship
+        E((Person - Dave))
+        -- FRIENDS -->
+        F((Person - Chris))  
+    end
+```
+Figure 1 - Relationship between Dave and Chris.
```
Comment (tb06904) [inline docs/user-guide/gaffer-basics/what-is-a-graph.md]: You can give a code block a title which might be preferable to this e.g. https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-a-title
```

File: docs/user-guide/gaffer-basics/what-is-a-graph.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    end
+```
+Figure 2 - Relationship between Dave and Chris. A potential relationship between Steve and Dave.
+
+
+## Key Terms
+
+| Term        | Description                          |
+| :---------- | :----------------------------------- |
+| Node        | A node is an entity within a graph  |
```
Comment (tb06904) [inline docs/user-guide/gaffer-basics/what-is-a-graph.md]: Is it worth mentioning vertex here too as that is widely used in Gaffer?

File: docs/user-guide/gaffer-basics/what-is-a-graph.md (hunk 1 above)
Comment (rj77259) [reply docs/user-guide/gaffer-basics/what-is-a-graph.md]: Wrapped the diagrams in example blocks instead. Title doesn't appear to work with mermaid.

File: docs/user-guide/gaffer-basics/what-is-a-graph.md (hunk 2 above)
Comment (rj77259) [reply docs/user-guide/gaffer-basics/what-is-a-graph.md]: Vertext added.

**Review by tb06904 — APPROVED** [review]

File: docs/user-guide/gaffer-basics/what-is-a-graph.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            E((Person - Dave))
+            -- FRIENDS -->
+            F((Person - Chris)) 
+            -- FRIENDS -->
+            B((Person - Steve)) 
+            B-. POTENTIAL RELATIONSHIP .->E;
+        end
+    ```
+
+## Key Terms
```
Comment (GCHQDeveloper314) [inline docs/user-guide/gaffer-basics/what-is-a-graph.md]: I think this duplicates what we have in the glossary under #399. Maybe we could wait till that is merged and then just add a link on this page pointing to that?

Comment (rj77259) [reply docs/user-guide/gaffer-basics/what-is-a-graph.md]: I will remove it for now.

Comment (GCHQDeveloper314) [reply docs/user-guide/gaffer-basics/what-is-a-graph.md]: Now the glossary page has been added, could we put a link to it in here?

**Review by GCHQDeveloper314 — APPROVED** [review]

**Pull request merged**